dev/devicelab/lib/framework/cocoon.dart

```diff
@@ -49,6 +52,17 @@ class Cocoon {
     return _commitSha;
   }
 
+  /// Parse the local repo for the current running branch.
+  String _readCommitBranch() {
```

Comment:
The way I would test this is by optionally injecting a tear-off Process.runSync, for example:

```
typedef ProcessRunSync = ProcessResult Function(String, List<String>, ...more stuff)

class Cocoon {
  Cocoon({..., @visibleForTesting this.processRunSync = Process.runSync }) ;

   final ProcessRunSync processRunSync;

}
```

Then you can provide your own callback for testing that returns an example output you observed.

Reply:
Thanks Jonah! Added tests around the git cli throwing issues.